src/backend/Level0/CHIPBackendLevel0.cc

```diff
@@ -1119,12 +1121,13 @@ void CHIPQueueLevel0::executeCommandList(ze_command_list_handle_t CommandList) {
     // Associate this event with the command list. Once the events are signaled,
     // CHIPEventMonitorLevel0 will destroy the command list
 
+    logDebug("assoc event {} w/ cmdlist", (void *)LastCmdListEvent);
     ((CHIPBackendLevel0 *)Backend)
         ->EventCommandListMap[(CHIPEventLevel0 *)LastCmdListEvent] =
         CommandList;
 
-    Status = zeCommandListAppendBarrier(CommandList, LastCmdListEvent->get(), 0,
-                                        nullptr);
+    Status = zeCommandListAppendBarrier(CommandList, LastCmdListEvent->peek(),
+                                        0, nullptr);
```

Comment:
If you change this from `get()` to `peek()` the refcount for this event is 0 when we begin tracking it. Once the event fires, it will try to decrement it from 0 to -1 which is bad.